src/ml_board/backend/restful_api/data_access.py

```diff
@@ -236,6 +240,70 @@ def get_experiment_config(self, grid_search_id: str, experiment_id: str, config_
         else:
             raise InvalidPathError(f"File path {requested_full_path} is not safe.")
 
+    def get_checkpoint_dict_epoch(self, grid_search_id: str, experiment_id: str, epoch: str):
+        """
+        Fetch all checkpoint resource pickle files from event storage
+        given the epoch, experiment ID & grid search ID.
+
+        :params:
+             grid_search_id (str): Grid Search ID
+             experiment_id (str): Experiment ID
+             epoch (str): Epoch number
+
+        :returns: List of Checkpoint files
+        """
+        requested_full_path = os.path.realpath(
+            os.path.join(self.top_level_logging_path, str(grid_search_id), str(experiment_id), str(epoch))
+        )
+
+        if FileDataAccess.is_safe_path(base_dir=self.top_level_logging_path, requested_path=requested_full_path):
+            response = []
+            checkpoints = []
+            files = FileDataAccess.get_checkpoint_files(requested_full_path, base_path=self.top_level_logging_path)
+            for file_num in range(len(files)):
+                split = os.path.normpath(files[file_num]).split(os.sep)
+                checkpoints.append(os.path.basename(split[3]).split(".")[0])
```

Comment:
is `split[3]` the filename here? In that case it makes more sense to use `split[-1]` in case the file_path changes. The hardcoded index of 3 can easily lead to side-effects, no?

Reply:
the split[3] is a filename, yes. The index path will not change because the index path is a fixed folder structure with the path as grid-search-id/experiment-id/epoch/checkpoint.pickle. As you said, we can change it to -1 but there are no side effects due to the fixed location of the checkpoint resource file in the event storage.